lib/devise/failure_app.rb

```diff
@@ -279,7 +279,7 @@ def root_path_defined?(context)
     end
 
     def rails_51_and_up?
-      Rails::VERSION::MAJOR >= 5 && Rails::VERSION::MINOR >= 1
+      Rails::VERSION::MAJOR >= 6 || (Rails::VERSION::MAJOR >= 5 && Rails::VERSION::MINOR >= 1)
```

Comment:
WDYT of using the `Gem::Version` comparator like we are doing [here](https://github.com/plataformatec/devise/blob/e3a00b27d19ba995891d7dd92394fe2900a789c2/lib/devise.rb#L305)?
So it would be something like this:
```ruby
Rails.gem_version >= Gem::Version.new("5.1")
```

Reply:
Fixed, thanks!